include/nbl/video/CThreadSafeGPUQueueAdapter.h

```diff
@@ -12,16 +12,23 @@ namespace nbl::video
 class CThreadSafeGPUQueueAdapter : public IGPUQueue
 {
     protected:
-        core::smart_refctd_ptr<IGPUQueue> originalQueue = nullptr;
+        core::smart_refctd_ptr<ILogicalDevice> logicalDevice = nullptr;
+        IGPUQueue* originalQueue = nullptr;
```

Comment:
original queue can be a smartpointer

Reply:
oh wait no it cant, my bad, I'll shut up

Reply:
resolved